pkg/kubelet/pod_workers.go

```diff
@@ -1177,6 +1182,9 @@ func (p *podWorkers) removeTerminatedWorker(uid types.UID) {
 		klog.V(4).InfoS("Pod has been terminated and is no longer known to the kubelet, remove all history", "podUID", uid)
 	}
 	delete(p.podSyncStatuses, uid)
+	if ch, ok := p.podUpdates[uid]; ok {
```

Comment:
What triggered adding this?

Reply:
Is this because we're creating a worker for the transient version of pod, but it is never closed?  And that leaves a pod worker running?

If so, we may want to have a better way of guaranteeing that all 3 "terminal" methods correctly do the same cleanup (extract it into a single function that assumes the lock i held)

Reply:
done